e2e/e2e_test.go

```diff
@@ -0,0 +1,163 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/utils/pointer"
+)
+
+const blockingPodName = "api-blocker-pod"
+
+var _ = Describe("e2e", func() {
+	var targetNode string
+
+	BeforeEach(func() {
+		// randomly pick a host (or let the scheduler do it by running the blocking pod)
+		// block the api port to make it go Ready false
+		node, err := makeNodeUnready(time.Minute * 5)
+		Expect(err).NotTo(HaveOccurred())
+		targetNode = node
+	})
+
+	AfterEach(func() {
+		removeAPIBlockingPod()
+	})
+	When("Node conditions meets the unhealthy criteria", func() {
+		It("Remediates a host", func() {
+			Expect(nhcRemediatesTheHost(targetNode, time.Minute*1)).To(Succeed())
+			Expect(nodeObjectIsRemoved(targetNode, time.Minute*1)).To(Succeed())
```

Comment:
1 minute might be too low, maybe we should set the timeout in PPC and use it

Reply:
yes, part of latest discussions. So an concrete suggestion for now?

Reply:
my suggestion would be to set some timeout value in the poison-pill-config CR, and use it here with few seconds buffer

Reply:
added. it workes good with the added timeouts.